language/stdlib/modules/libra_writeset_manager.move

```diff
@@ -0,0 +1,55 @@
+module LibraWriteSetManager {
+    use 0x0::LibraAccount;
+    use 0x0::Hash;
+    use 0x0::Transaction;
+    use 0x0::LibraSystem;
+
+    resource struct T {
+        sequence_number: u64,
+        history_events: LibraAccount::EventHandle<Self::UpgradeEvent>,
+    }
+
+    struct UpgradeEvent {
+        writeset_payload: vector<u8>,
+    }
+
+    public fun initialize() {
+        Transaction::assert(Transaction::sender() == 0xA550C18, 1);
+
+        move_to_sender<T>(T {
+            sequence_number: 0,
+            history_events: LibraAccount::new_event_handle<Self::UpgradeEvent>(),
+        });
+    }
+
+    fun prologue(
+        writeset_sequence_number: u64,
+        writeset_public_key: vector<u8>,
+    ) acquires T {
+        let sender = Transaction::sender();
+        Transaction::assert(sender == 0xA550C18, 33);
+
+        let association_auth_key = LibraAccount::authentication_key(sender);
+
+        let t_ref = borrow_global<T>(0xA550C18);
+        Transaction::assert(writeset_sequence_number >= t_ref.sequence_number, 3);
+
+        // TODO: We don't use SEQUENCE_NUM_TOO_NEW(4) for now as it will be ignored by the VMValidator.
```

Comment:
What are the potential consequences of this?

Reply:
Currently the writeset prologue is reusing the error code as the regular transaction prologue. That means it will have the same sequence number too new/old error code. In the VMValidator, we intensionally let the SEQUENCE_NUMBER_TOO_NEW through because of the false negative concern, meaning that a transaction with the right sequence number has already been submitted but not readable from VMValidator at this point (due to network delay, waiting for blocks to be committed, etc). In the `WriteSet` example however, I don't think such false negative is going to happen that often, so I intensionally didn't use `SEQUENCE_NUMBER_TOO_NEW` error so that such transaction will be rejected at VMValidation time.

Reply:
Ok, interesting. Perhaps we should either explain the full context in a comment (like you did here), or kill the comment? I  think that the comment as-is is a bit confusing.